### ISSUE TITLE
Rename script, .desktop, icon, etc spyder3 => spyder4

### ISSUE THREAD

**Issue by lucatrv** [spyder-ide/spyder#11835]
Searching `spyder3` in this repository still finds script, .desktop, icon, etc file names, they should be renamed `spyder4`.

**Comment by ccordoba12** [comment]
That's a requirement of Linux distros: the 3 in `spyder3` refers to Python 3, not to Spyder 3.

**Comment by lucatrv** [comment]
IMHO that's not a _requirement_. You could rename it `spyder`, or `python3-spyder`, or `python3-spyder4`, etc. `spyder3` is misleading.

**Comment by ccordoba12** [comment]
Every other package on Linux uses the same convention (e.g. ipython vs. ipython3), so we're just merely following the trend.

If you find this misleading, please use Anaconda, which doesn't have this problem (i.e. our executable script is named spyder in Python 2 and 3).

**Comment by lucatrv** [comment]
That convention depends on the actual distribution. For instance, Arch Linux provides a package [`ipython`](https://www.archlinux.org/packages/community/any/ipython/) for Python 3 (which includes executables `usr/bin/ipython` and `usr/bin/ipython3`), and a package [`ipython2`](https://www.archlinux.org/packages/community/any/ipython2/) for Python 2 (which includes executable `usr/bin/ipython2`). Fedora, which has already moved away from Python2 (as most other distributions are doing) provides a main package [`ipython`](https://apps.fedoraproject.org/packages/ipython) which depends on [`python3-ipython`](https://apps.fedoraproject.org/packages/python3-ipython/) (and others).

See also the `find_entry_points` function in [`setupbase.py`](https://github.com/ipython/ipython/blob/master/setupbase.py). On Windows `ipython` installs both `ipython.exe` and `ipython3.exe`.

Since Spyder requires Python 3, in your mainline package you could define only the `spyder` script, or both `spyder` and `spyder3` scripts if you will (it would behave like `ipython`). Then of course each Linux distribution mantainer can apply a customization to its distributed package if needed.

Just my 2 cents of course.

**Comment by ccordoba12** [comment]
Spyder still works for Python 2 and 3.

**Comment by lucatrv** [comment]
Oh sorry, I thought I read that Spyder 4 deprecated Python 2 support, never mind. In this case you could define both `spyder` and `spyder3` for Python 3, and `spyder2` for Python 2. No doubts that nowadays most users have Python 3 installed, and would expect to run Spyder with command `spyder`. I have Python 3 installed, and I can run IPython both with commands `ipython` and `ipython3`.

**Comment by PanderMusubi** [comment]
Quick question, since January 1 2020, Python 2 has been deprecated. The files `~/.local/bin/spyder3` and `~/.local/bin/spyder` are identical. When I start `spyder` from the command line, the filename completion with TAB doesn't work, only after typing a 3 or a space. Is the request to remove `/.local/bin/spyder3` good to create an issue for?

**Comment by ccordoba12** [comment]
> Two related questions, if they are suitable for an issue or not:

I don't what version you installed, but that's already done in Spyder 6:

https://github.com/spyder-ide/spyder/blob/e95637e9a42f50b5a58b76d15a3d5073416b3b9d/scripts/spyder#L1-L3
